packages/code-studio/vite.config.ts

```diff
@@ -40,6 +40,17 @@ export default defineConfig(({ mode }) => {
       target: `http://localhost:${port}/src/styleguide/index.html`,
       rewrite: () => '',
     },
+    // proxy the websocket requests, allows tunneling to work with a single port
+    '^/arrow.*': {
+      target: env.VITE_PROXY_URL,
+      changeOrigin: true,
+      ws: true,
+    },
+    '^/io.deephaven.*': {
```

Comment:
This regex isn't quite right, you should be escaping the periods, e.g.:
```suggestion
    '^/io\.deephaven\..*': {
```
But aside from that - gRPC could have any path, we should be doing a negative lookup for any of the routes we reserve instead, e.g.:
```
'^(?!/ide).*'
```
```

Reply:
Can't do negative lookup. vite serves from a whole wack of routes. gRPC could have any path, but will it?

![image](https://user-images.githubusercontent.com/1576283/192360777-8d2778f0-6613-43f3-a465-5b1a1a7d1195.png)

Reply:
Okay, seems like just `arrow.` and `io.deephaven.` are the only two we (currently) need to worry about.
However should still make the changes I suggested to have the correct regexes, `'^/arrow\..*'` and `'^/io\.deephaven\..*'`